test/e2e_node/lifecyclehooks/prestop-hook.go

```diff
@@ -0,0 +1,141 @@
+        podClient = e2epod.NewPodClient(f)
+    })
+ 
+    ginkgo.It("should run PreStop hook basic test case and verify execution ", func(ctx context.Context) {
+ 
+        TestName("Hooks:PreStop Hook Basic Test")
+        TestStep("When you have a PreStop hook defined on your container, it will execute before the container is terminated within grace period.")
+ 
+        volumeName := "shared-data"
+        mountPath := "/tmp/prestop-test-logs"
+ 
+        // Define the lifecycle with the PreStop hook
+        lifecycle := &v1.Lifecycle{
+            PreStop: &v1.LifecycleHandler{
+                Exec: &v1.ExecAction{
+                    Command: []string{
+                        "sh",
+                        "-c",
+                        fmt.Sprintf("echo 'PreStop Hook Executed' > %s/prestop_hook_executed; sleep 10", mountPath),
+                        //"echo 'PreStop Hook Executed' > /tmp/prestop-test-logs/prestop_hook_executed",
+                    },
+                },
+            },
+        }
+ 
+        TestStep("Imagine You Have a Pod with the Following Specification")
+ 
+        // Define the pod with the emptyDir volume and appropriate security context
+        podWithHook := &v1.Pod{
+            ObjectMeta: metav1.ObjectMeta{
+                Name: "prestop-pod",
+            },
+            Spec: v1.PodSpec{
+                Containers: []v1.Container{
+                    {
+                        Name:    "prestop-container",
+                        Image:   imageutils.GetE2EImage(imageutils.Agnhost),
+                        Args:    []string{"pause"},
+                        Lifecycle: lifecycle,
+                        VolumeMounts: []v1.VolumeMount{
+                            {
+                                Name:      volumeName,
+                                MountPath: mountPath,
+                            },
+                        },
+                        SecurityContext: &v1.SecurityContext{
+                            RunAsNonRoot:             boolPtr(true),
+                            RunAsUser:                int64Ptr(1000),
+                            RunAsGroup:               int64Ptr(1000),
+                            AllowPrivilegeEscalation: boolPtr(false),
+                            Capabilities: &v1.Capabilities{
+                                Drop: []v1.Capability{"ALL"},
+                            },
+                            SeccompProfile: &v1.SeccompProfile{
+                                Type: v1.SeccompProfileTypeRuntimeDefault,
+                            },
+                        },
+                    },
+                },
+                Volumes: []v1.Volume{
+                    {
+                        Name: volumeName,
+                        VolumeSource: v1.VolumeSource{
+                            EmptyDir: &v1.EmptyDirVolumeSource{
+                                Medium: v1.StorageMediumDefault,
+                            },
```

Comment:
Maybe use a host path volume instead? so that you don't need to know the kubelet's volume structure

Reply:
You'll need to add `f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged` on line 23 to use a hostpath volume